app/services/proofing/resolution/plugins/instant_verify_residential_address_plugin.rb

```diff
@@ -24,8 +24,36 @@ def call(
         end
 
         def proofer
-          @proofer ||=
-            if IdentityConfig.store.proofer_mock_fallback
+          @proofer ||= begin
+            # Historically, proofer_mock_fallback has controlled whether we
+            # use mock implementations of the Resolution and Address proofers
+            # (true = use mock, false = don't use mock).
+            # We are transitioning to a place where we will have separate
+            # configs for both. For the time being, we want to keep support for
+            # proofer_mock_fallback here. This can be removed after this code
+            # has been deployed and configs have been updated in all relevant
+            # environments.
+
+            old_config_says_mock = IdentityConfig.store.proofer_mock_fallback
+            old_config_says_iv = !old_config_says_mock
+            new_config_says_mock =
+              IdentityConfig.store.idv_resolution_default_vendor == :mock
+            new_config_says_iv =
+              IdentityConfig.store.idv_resolution_default_vendor == :instant_verify
+
+            proofer_type =
+              if new_config_says_mock && old_config_says_iv
+                # This will be the case immediately after deployment, when
+                # environment configs have not been updated. We need to
+                # fall back to the old config here.
+                :instant_verify
+              elsif new_config_says_iv
+                :instant_verify
+              else
+                :mock
+              end
```

Comment:
I'm a fan of really expressive code like this for this sort of thing.

I think customarily we'd argue that `old_config_says_mock` is unneeded as a variable, and the first two cases of this conditional could be combined. But in temporary code handling something a little complicated to reason about? This makes it really easy to understand.